### ISSUE TITLE
Critical loss of accuracy when converting from MobileNet V2 model to tflite

### ISSUE THREAD

**Issue by jenny-pixel** [tensorflow/tensorflow#47761]
### 1. System information

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installation (pip package or built from source): pip package
- TensorFlow library (version, if pip package or github SHA, if built from source): 1.13.1 (and 2.0.0)

### 3. Failure after conversion
If the conversion is successful, but the generated model is wrong, then state what is wrong:
- Model produces wrong results and/or has lesser accuracy.

I have retrained a Mobilenet_V3_1.0 model using Deeplabcut. When I convert the model to tflite, the accuracy drops to <0.01 whereas the accuracy was previously >0.9. If I do the same, but retrain a resnet_50 model, the tflite conversion works correctly, giving pretty much the same accuracy as the original.

I also found that the accuracy of the tflite model is inversely proportional to the accuracy (or training iteration) of the original retrained model. So the more training iterations and the better the accuracy of the model, the worse the accuracy of the tflite model.

Here's the code I used to convert the graph (which works fine on a model retrained from a resnet base)

video =<path to video>
model_file<path to .pb model file>

cap = cv2.VideoCapture(video)
ret, frame = cap.read()
with tf.io.gfile.GFile(model_file, "rb") as f:
        graph_def = tf.GraphDef()
        graph_def.ParseFromString(f.read())

graph = tf.Graph()
with graph.as_default():
	tf.import_graph_def(graph_def, name="DLC")

graph.finalize()
op_names = [str(op.name) for op in graph.get_operations()]

output_nodes = [op_names[-1], op_names[-2]]
output_nodes = [on.replace("DLC/", "") for on in output_nodes]

converter = tf.lite.TFLiteConverter.from_frozen_graph(
                model_file,
                ["Placeholder"],
                output_nodes,
                input_shapes={"Placeholder": [1, frame.shape[0], frame.shape[1], 3]},
            )
tflite_model = converter.convert()
with open('model.tflite', 'wb') as f:
  f.write(tflite_model)

And then for calling the tflite model: 
tflite_interpreter = tf.lite.Interpreter(model_content=tflite_model)
tflite_interpreter.allocate_tensors()
inputs = tflite_interpreter.get_input_details()
outputs = tflite_interpreter.get_output_details()

ret, frame = cap.read()
im = np.expand_dims(frame, axis=0).astype(np.float32)
tflite_interpreter.set_tensor(
		        inputs[0]["index"],
		        im
		    )
tflite_interpreter.invoke()

outputs_np = [
		           tflite_interpreter.get_tensor(outputs[0]["index"]),
		           tflite_interpreter.get_tensor(outputs[1]["index"]),
		        ]

In short - I'm not sure why the conversion would work fine for a retrained resnet model and not a mobilenet model? Any ideas?
Thanks!

**Comment by abattery** [comment]
Could you try your TFLite conversion script from the recent TF version? FYI, you can bring your TF v1 model file from the newer TFLiteConverter.

**Comment by jenny-pixel** [comment]
The most recent version of TFLiteConverter doesn't have a "from_frozen_graph" option...

**Comment by abattery** [comment]
Actually, you can find the frozen graph option under the `tf.compat.v1.lite.TFLiteConverter`.

**Comment by jenny-pixel** [comment]
Hi, unfortunately I still have the same trouble. The code (conversion etc) works well for my model when trained using a resnet backbone, but not when I use mobilenet. The error also increase the more training iterations I complete.

**Comment by xhae** [comment]
It looks like you're doing classification with video stream if I read the code correctly, any reason not doing this with the static image?
One rough idea I have is that accuracy>0.9(or >0.99 based on your other issue: https://github.com/DeepLabCut/DeepLabCut-live/issues/43) sounds awesome, but there's not enough resource on how did you train and evaluate.
